### DISCUSSION TITLE
Parameter randomization suggestions

### DISCUSSION THREAD

**Discussion by x-iso** [kernicPanel/bitwig-randomizer#11] [Ideas]
There are some ways parameter randomization feature could get more useful:

- Note driven randomization trigger. If possible to track incoming notes for currently focused device, would be good to have option for triggering random with each new note. 

- Ability to exclude some parameters from randomization, by specific parameter or by whole pages, and option to only randomize preset pages. Most presets would include parameters like Output level, which you'd normally don't want to randomize. I don't know what ways would be possible to exclude parameters, but I think using matching keywords could work if selection isn't possible. Another way could be by choosing specific slot in Remote control page, since Output for example is 99% of the time located at 8th slot, this method can work out too.

- New values fade in time. So that when random is triggered, values move from current position to new one in given interval starting from 0ms to some arbitrary range.

**Comment by kernicPanel** [comment]
Thanks for your suggestions @x-iso 
But I fear you won't like my answers, as this extension purpose does not cover "live" randomization.
The point of this is to help to find sounds manually.

For the Note driven randomization trigger and the New values fade in time: I think this would be more likely done through modulators, or an FX grid.
For the fun, I'll try to propose you something if you want.

For the parameters exclusion, I already excluded parameters named "Output" (kind of buggy, but it works at least for the current parameter page). 
For now, I'm not able to only randomize preset pages, so a workaround might be to build a page in which you only put parameters you want to randomize.
Of course, if you have any code suggestion to improve the exclusion, feel free to put them here, or open a PR.

**Comment by x-iso** [comment]
I see. Well, I do have a Grid patch that's designed for live randomization, but I'm using it with Generic contriller script over virtual midi port. Downside is that it can't read current values of remote controls (but relative control takeover mode kind of takes care of this), and it can only target current remote control page. I hoped that it would be possible to at least combine the two in some way to overlap advantages. As far as I understand form your answer it's not possible to (or you don't know how to) get note trigs from focused device just with script alone? If it was possible to eliminate usage of virtual midi for patch and script to work together, that would be already much more user friendly experience.

**Comment by kernicPanel** [comment]
For now, I don't really have time to dig into this, but I like the idea, thanks!
